q2_feature_table/_split.py

```diff
@@ -0,0 +1,28 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2023, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import biom
+import qiime2
+
+
+def split(table: biom.Table,
+          metadata: qiime2.CategoricalMetadataColumn,
+          filter_empty_features: bool = True) -> biom.Table:
+
+    metadata_df = metadata.drop_missing_values().to_dataframe()
```

Comment:
So this line of code will drop any values within the given md column that don't have a value - should this be distinctly different from the intended functionality of the `filter_empty_features` parameter? The way I'm understanding that parameter, it should do the same thing (if enabled). I tested this out locally and received the same results when this flag was both enabled and disabled, using the following data:
```
sample-id    body-site
L1S8             NaN
L1S57            NaN
L1S76            NaN
L1S105           NaN
L2S155     left palm
L2S175     left palm
L2S204     left palm
L2S222     left palm
L3S242    right palm
L3S294    right palm
L3S313    right palm
L3S341    right palm
L3S360    right palm
L5S104        tongue
L5S155        tongue
L5S174        tongue
L5S203        tongue
L5S222        tongue
L1S140           NaN
L1S208           NaN
L1S257           NaN
L1S281           NaN
L2S240     left palm
L2S309     left palm
L2S357     left palm
L2S382     left palm
L3S378    right palm
L4S63     right palm
L4S112    right palm
L4S137    right palm
L5S240        tongue
L6S20         tongue
L6S68         tongue
L6S93         tongue
```
What I got back were three tables (for 'left palm', 'right palm', and 'tongue', respectively) - I would expect to get back a fourth table when `filter_empty_features` isn't enabled that would include all of the features without an associated group in this md column, is that what you intended?

Reply:
@lizgehret, this is defining the behavior for features that would have a total frequency of zero in a split feature table (see my `test_drop_features_w_zero_count` test case). So, for example, if an ASV is only present in `tongue` samples in the input table and `filter_empty_features == False`, that ASV will be included in the output `left palm` and `right palm` tables but it's total frequency in each will be zero (since it wasn't observed in either of those sample types in the input table). If `filter_empty_features == True`, it would not be included in the `left palm` and `right palm` tables. 

Does that make sense? Let me know if you have suggestions for a better parameter name or documentation to make this more clear.

Reply:
Ahh, yes that makes sense - thanks for clarifying! After re-reading your parameter descriptions, I think this seems reasonably clear.